stripe/src/main/java/com/stripe/android/paymentsheet/DefaultPrimaryButton.kt

```diff
@@ -0,0 +1,95 @@
+package com.stripe.android.paymentsheet
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import androidx.core.view.isVisible
+import com.stripe.android.R
+import com.stripe.android.databinding.DefaultPrimaryButtonBinding
+import com.stripe.android.paymentsheet.model.ViewState
+import com.stripe.android.paymentsheet.ui.PrimaryButton
+import com.stripe.android.paymentsheet.ui.PrimaryButtonAnimator
+
+internal open class DefaultPrimaryButton @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : PrimaryButton(context, attrs, defStyleAttr) {
+    private var viewState: ViewState? = null
+    private val animator = PrimaryButtonAnimator(context)
+
+    internal val viewBinding = DefaultPrimaryButtonBinding.inflate(
+        LayoutInflater.from(context),
+        this
+    )
+
+    private val confirmedIcon = viewBinding.confirmedIcon
+
+    init {
+        setBackgroundResource(R.drawable.stripe_paymentsheet_buy_button_default_background)
+
+        isClickable = true
+        isEnabled = false
+    }
+
+    fun setLabelText(text: String){
+        viewBinding.label.text = text
+    }
+
+    fun setReady(){
+        viewBinding.confirmingIcon.isVisible = false
+    }
+
+    fun setConfirm(){
+        viewBinding.lockIcon.isVisible = false
+        viewBinding.confirmingIcon.isVisible = true
+
+        viewBinding.label.text = resources.getString(
+            R.string.stripe_paymentsheet_pay_button_processing
+        )
+    }
+
+    fun setCompleted(onAnimationEnd: () -> Unit){
```

Comment:
Thinking out loud: instead of passing the lambda here, can we set it earlier?

```
val addButton = viewBinding.addButton
addButton.onAnimationEnd = { ... }
```

Reply:
I was just thinking about this one.  I think this is an issue because if the activity finishes before the animation completes we might have issues.  I need to rethink this a bit.

Reply:
Yeah worth examining but I think we can find a way to handle it.